language/move-prover/boogie-backend/src/bytecode_translator.rs

```diff
@@ -1199,6 +1199,28 @@ impl<'env> ModuleTranslator<'env> {
                         self.track_return(fun_target, *i, srcs[0]);
                     }
                     TraceAbort => self.track_abort(fun_target, &str_local(srcs[0])),
+                    EmitEvent => {
+                        let msg = srcs[0];
+                        let handle = srcs[1];
+                        emit!(
+                            self.writer,
+                            "$es := ${}ExtendEventStore($es, ",
+                            if srcs.len() > 2 { "Cond" } else { "" }
+                        );
+                        emit!(
+                            self.writer,
+                            "$SelectField({}, $Event_EventHandle_guid), {}",
```

Comment:
I think that this should be replaced by `$SelectField($Dereference({}), $Event_EventHandle_guid), {}",`.

Reply:
Yes, if the `{}` substitution is a `&mut` in the Move code, then it is now a $Mutation and the $Dereference needs to be inserted.